# IMDB Sentiment Classification: Embedding + LSTM

In [1]:
import os
os.environ["KMP_DUPLICATE_LIB_OK"] = "True"

from pathlib import Path

import matplotlib
matplotlib.use("Agg")
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import seaborn as sns
from sklearn.metrics import classification_report, confusion_matrix
from tensorflow import keras
from tensorflow.keras.preprocessing.sequence import pad_sequences

project_root = Path.cwd().parent if Path.cwd().name == "notebooks" else Path.cwd()
results_dir = project_root / "results"
models_dir = project_root / "models"
results_dir.mkdir(exist_ok=True)
models_dir.mkdir(exist_ok=True)

VOCAB_SIZE = 10000
MAX_LEN = 200

print("Imports ready")
print("Project root:", project_root)
print("VOCAB_SIZE:", VOCAB_SIZE, "MAX_LEN:", MAX_LEN)

Matplotlib is building the font cache; this may take a moment.


Imports ready
Project root: /Users/yuvrajgole/Documents/DL Experiments/DL Experiment 5/nlp-imdb-lstm
VOCAB_SIZE: 10000 MAX_LEN: 200


## Section 1: Load Dataset

In [2]:
(X_train, y_train), (X_test, y_test) = keras.datasets.imdb.load_data(num_words=VOCAB_SIZE)

print("len(X_train):", len(X_train))
print("len(X_test):", len(X_test))

       0/17464789 ━━━━━━━━━━━━━━━━━━━━ 0s 0s/step

   32768/17464789 ━━━━━━━━━━━━━━━━━━━━ 33s 2us/step

   49152/17464789 ━━━━━━━━━━━━━━━━━━━━ 1:42 6us/step

   81920/17464789 ━━━━━━━━━━━━━━━━━━━━ 1:54 7us/step

  114688/17464789 ━━━━━━━━━━━━━━━━━━━━ 1:31 5us/step

  147456/17464789 ━━━━━━━━━━━━━━━━━━━━ 1:19 5us/step

  163840/17464789 ━━━━━━━━━━━━━━━━━━━━ 1:18 5us/step

  196608/17464789 ━━━━━━━━━━━━━━━━━━━━ 1:10 4us/step

  229376/17464789 ━━━━━━━━━━━━━━━━━━━━ 1:05 4us/step

  262144/17464789 ━━━━━━━━━━━━━━━━━━━━ 1:00 4us/step

  311296/17464789 ━━━━━━━━━━━━━━━━━━━━ 54s 3us/step 

  376832/17464789 ━━━━━━━━━━━━━━━━━━━━ 47s 3us/step

  442368/17464789 ━━━━━━━━━━━━━━━━━━━━ 42s 3us/step

  507904/17464789 ━━━━━━━━━━━━━━━━━━━━ 38s 2us/step

  573440/17464789 ━━━━━━━━━━━━━━━━━━━━ 35s 2us/step

  663552/17464789 ━━━━━━━━━━━━━━━━━━━━ 31s 2us/step

  770048/17464789 ━━━━━━━━━━━━━━━━━━━━ 28s 2us/step

  901120/17464789 ━━━━━━━━━━━━━━━━━━━━ 25s 2us/step

 1048576/17464789 ━━━━━━━━━━━━━━━━━━━━ 22s 1us/step

 1212416/17464789 ━━━━━━━━━━━━━━━━━━━━ 19s 1us/step

 1392640/17464789 ━━━━━━━━━━━━━━━━━━━━ 17s 1us/step

 1622016/17464789 ━━━━━━━━━━━━━━━━━━━━ 15s 1us/step

 1867776/17464789 ━━━━━━━━━━━━━━━━━━━━ 13s 1us/step

 2170880/17464789 ━━━━━━━━━━━━━━━━━━━━ 11s 1us/step

 2531328/17464789 ━━━━━━━━━━━━━━━━━━━━ 10s 1us/step

 2924544/17464789 ━━━━━━━━━━━━━━━━━━━━ 8s 1us/step 

 3416064/17464789 ━━━━━━━━━━━━━━━━━━━━ 7s 1us/step

 3973120/17464789 ━━━━━━━━━━━━━━━━━━━━ 6s 0us/step

 4620288/17464789 ━━━━━━━━━━━━━━━━━━━━ 5s 0us/step

 5390336/17464789 ━━━━━━━━━━━━━━━━━━━━ 4s 0us/step

 6275072/17464789 ━━━━━━━━━━━━━━━━━━━━ 3s 0us/step

 6930432/17464789 ━━━━━━━━━━━━━━━━━━━━ 3s 0us/step

 7602176/17464789 ━━━━━━━━━━━━━━━━━━━━ 2s 0us/step

 8273920/17464789 ━━━━━━━━━━━━━━━━━━━━ 2s 0us/step

 8929280/17464789 ━━━━━━━━━━━━━━━━━━━━ 2s 0us/step

 9617408/17464789 ━━━━━━━━━━━━━━━━━━━━ 1s 0us/step

10272768/17464789 ━━━━━━━━━━━━━━━━━━━━ 1s 0us/step

10944512/17464789 ━━━━━━━━━━━━━━━━━━━━ 1s 0us/step

11599872/17464789 ━━━━━━━━━━━━━━━━━━━━ 1s 0us/step

12288000/17464789 ━━━━━━━━━━━━━━━━━━━━ 1s 0us/step

12943360/17464789 ━━━━━━━━━━━━━━━━━━━━ 0s 0us/step

13631488/17464789 ━━━━━━━━━━━━━━━━━━━━ 0s 0us/step

14303232/17464789 ━━━━━━━━━━━━━━━━━━━━ 0s 0us/step

14974976/17464789 ━━━━━━━━━━━━━━━━━━━━ 0s 0us/step

15646720/17464789 ━━━━━━━━━━━━━━━━━━━━ 0s 0us/step

16318464/17464789 ━━━━━━━━━━━━━━━━━━━━ 0s 0us/step

17006592/17464789 ━━━━━━━━━━━━━━━━━━━━ 0s 0us/step

17464789/17464789 ━━━━━━━━━━━━━━━━━━━━ 3s 0us/step


len(X_train): 25000
len(X_test): 25000


## Section 2: Text Preprocessing (Padding)

In [3]:
X_train = pad_sequences(X_train, maxlen=MAX_LEN, padding="post", truncating="post")
X_test = pad_sequences(X_test, maxlen=MAX_LEN, padding="post", truncating="post")

print("X_train.shape:", X_train.shape)
print("X_test.shape:", X_test.shape)

X_train.shape: (25000, 200)
X_test.shape: (25000, 200)


## Section 3: Build Embedding + LSTM Model

In [4]:
model = keras.Sequential(
    [
        keras.layers.Input(shape=(MAX_LEN,)),
        keras.layers.Embedding(VOCAB_SIZE, 128),
        keras.layers.LSTM(64, dropout=0.2, recurrent_dropout=0.2),
        keras.layers.Dense(32, activation="relu"),
        keras.layers.Dropout(0.3),
        keras.layers.Dense(1, activation="sigmoid"),
    ]
)
model.compile(
    optimizer="adam",
    loss="binary_crossentropy",
    metrics=["accuracy"],
)
model.summary()

Model: "sequential"

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┓
┃ Layer (type)                    ┃ Output Shape           ┃       Param # ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━┩
│ embedding (Embedding)           │ (None, 200, 128)       │     1,280,000 │
├─────────────────────────────────┼────────────────────────┼───────────────┤
│ lstm (LSTM)                     │ (None, 64)             │        49,408 │
├─────────────────────────────────┼────────────────────────┼───────────────┤
│ dense (Dense)                   │ (None, 32)             │         2,080 │
├─────────────────────────────────┼────────────────────────┼───────────────┤
│ dropout (Dropout)               │ (None, 32)             │             0 │
├─────────────────────────────────┼────────────────────────┼───────────────┤
│ dense_1 (Dense)                 │ (None, 1)              │            33 │
└─────────────────────────────────┴────────────────────────┴───────────────┘

 Total params: 1,331,521 (5.08 MB)

 Trainable params: 1,331,521 (5.08 MB)

 Non-trainable params: 0 (0.00 B)

## Section 4: Train the Model

In [5]:
history = model.fit(
    X_train,
    y_train,
    validation_split=0.2,
    epochs=5,
    batch_size=64,
    verbose=1,
)

Epoch 1/5


  1/313 ━━━━━━━━━━━━━━━━━━━━ 8:13 2s/step - accuracy: 0.4688 - loss: 0.6929

  3/313 ━━━━━━━━━━━━━━━━━━━━ 13s 43ms/step - accuracy: 0.5156 - loss: 0.6914

  5/313 ━━━━━━━━━━━━━━━━━━━━ 13s 42ms/step - accuracy: 0.5094 - loss: 0.6896

  7/313 ━━━━━━━━━━━━━━━━━━━━ 12s 42ms/step - accuracy: 0.4911 - loss: 0.6929

  9/313 ━━━━━━━━━━━━━━━━━━━━ 12s 42ms/step - accuracy: 0.4878 - loss: 0.6966

 11/313 ━━━━━━━━━━━━━━━━━━━━ 12s 42ms/step - accuracy: 0.4872 - loss: 0.6967

 13/313 ━━━━━━━━━━━━━━━━━━━━ 12s 42ms/step - accuracy: 0.4856 - loss: 0.6966

 15/313 ━━━━━━━━━━━━━━━━━━━━ 12s 42ms/step - accuracy: 0.4875 - loss: 0.6961

 17/313 ━━━━━━━━━━━━━━━━━━━━ 12s 42ms/step - accuracy: 0.4917 - loss: 0.6957

 19/313 ━━━━━━━━━━━━━━━━━━━━ 12s 42ms/step - accuracy: 0.4951 - loss: 0.6954

 21/313 ━━━━━━━━━━━━━━━━━━━━ 12s 42ms/step - accuracy: 0.4903 - loss: 0.6952

 23/313 ━━━━━━━━━━━━━━━━━━━━ 12s 42ms/step - accuracy: 0.4966 - loss: 0.6949

 25/313 ━━━━━━━━━━━━━━━━━━━━ 12s 42ms/step - accuracy: 0.4981 - loss: 0.6947

 27/313 ━━━━━━━━━━━━━━━━━━━━ 12s 42ms/step - accuracy: 0.4965 - loss: 0.6948

 29/313 ━━━━━━━━━━━━━━━━━━━━ 11s 42ms/step - accuracy: 0.4935 - loss: 0.6950

 31/313 ━━━━━━━━━━━━━━━━━━━━ 11s 42ms/step - accuracy: 0.4955 - loss: 0.6948

 33/313 ━━━━━━━━━━━━━━━━━━━━ 11s 42ms/step - accuracy: 0.4915 - loss: 0.6948

 35/313 ━━━━━━━━━━━━━━━━━━━━ 11s 42ms/step - accuracy: 0.4915 - loss: 0.6947

 37/313 ━━━━━━━━━━━━━━━━━━━━ 11s 42ms/step - accuracy: 0.4916 - loss: 0.6946

 39/313 ━━━━━━━━━━━━━━━━━━━━ 11s 42ms/step - accuracy: 0.4920 - loss: 0.6945

 41/313 ━━━━━━━━━━━━━━━━━━━━ 11s 42ms/step - accuracy: 0.4878 - loss: 0.6945

 43/313 ━━━━━━━━━━━━━━━━━━━━ 11s 42ms/step - accuracy: 0.4891 - loss: 0.6944

 45/313 ━━━━━━━━━━━━━━━━━━━━ 11s 42ms/step - accuracy: 0.4913 - loss: 0.6943

 47/313 ━━━━━━━━━━━━━━━━━━━━ 11s 42ms/step - accuracy: 0.4920 - loss: 0.6943

 49/313 ━━━━━━━━━━━━━━━━━━━━ 11s 42ms/step - accuracy: 0.4920 - loss: 0.6943

 51/313 ━━━━━━━━━━━━━━━━━━━━ 11s 42ms/step - accuracy: 0.4939 - loss: 0.6943

 53/313 ━━━━━━━━━━━━━━━━━━━━ 10s 42ms/step - accuracy: 0.4962 - loss: 0.6941

 55/313 ━━━━━━━━━━━━━━━━━━━━ 10s 42ms/step - accuracy: 0.4972 - loss: 0.6941

 57/313 ━━━━━━━━━━━━━━━━━━━━ 10s 42ms/step - accuracy: 0.4967 - loss: 0.6941

 59/313 ━━━━━━━━━━━━━━━━━━━━ 10s 42ms/step - accuracy: 0.4952 - loss: 0.6941

 61/313 ━━━━━━━━━━━━━━━━━━━━ 10s 42ms/step - accuracy: 0.4959 - loss: 0.6940

 63/313 ━━━━━━━━━━━━━━━━━━━━ 10s 42ms/step - accuracy: 0.4958 - loss: 0.6940

 65/313 ━━━━━━━━━━━━━━━━━━━━ 10s 42ms/step - accuracy: 0.4942 - loss: 0.6940

 67/313 ━━━━━━━━━━━━━━━━━━━━ 10s 42ms/step - accuracy: 0.4942 - loss: 0.6940

 69/313 ━━━━━━━━━━━━━━━━━━━━ 10s 42ms/step - accuracy: 0.4943 - loss: 0.6940

 71/313 ━━━━━━━━━━━━━━━━━━━━ 10s 42ms/step - accuracy: 0.4963 - loss: 0.6940

 73/313 ━━━━━━━━━━━━━━━━━━━━ 10s 42ms/step - accuracy: 0.4959 - loss: 0.6940

 75/313 ━━━━━━━━━━━━━━━━━━━━ 9s 42ms/step - accuracy: 0.4975 - loss: 0.6939 

 77/313 ━━━━━━━━━━━━━━━━━━━━ 9s 42ms/step - accuracy: 0.4972 - loss: 0.6939

 79/313 ━━━━━━━━━━━━━━━━━━━━ 9s 42ms/step - accuracy: 0.4970 - loss: 0.6939

 81/313 ━━━━━━━━━━━━━━━━━━━━ 9s 42ms/step - accuracy: 0.4967 - loss: 0.6939

 83/313 ━━━━━━━━━━━━━━━━━━━━ 9s 42ms/step - accuracy: 0.4964 - loss: 0.6939

 85/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.4972 - loss: 0.6939

 87/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.4962 - loss: 0.6939

 89/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.4981 - loss: 0.6938

 91/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.5003 - loss: 0.6937

 93/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.5012 - loss: 0.6937

 95/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.5025 - loss: 0.6935

 97/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.5021 - loss: 0.6935

 99/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.5025 - loss: 0.6935

101/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.5015 - loss: 0.6935

103/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.5024 - loss: 0.6935

105/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.5028 - loss: 0.6934

107/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.5031 - loss: 0.6934

109/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.5026 - loss: 0.6933

111/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.5034 - loss: 0.6933

113/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.5024 - loss: 0.6934

115/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.5022 - loss: 0.6934

117/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.5035 - loss: 0.6933

119/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.5035 - loss: 0.6933

121/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.5049 - loss: 0.6933

123/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.5046 - loss: 0.6933

125/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.5061 - loss: 0.6932

127/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.5080 - loss: 0.6931

129/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.5093 - loss: 0.6930

131/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.5088 - loss: 0.6931

133/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.5090 - loss: 0.6930

135/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.5094 - loss: 0.6929

137/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.5090 - loss: 0.6930

139/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.5089 - loss: 0.6929

141/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.5088 - loss: 0.6929

143/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.5096 - loss: 0.6928

145/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.5099 - loss: 0.6927

147/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.5101 - loss: 0.6926

149/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.5108 - loss: 0.6926

151/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.5115 - loss: 0.6923

153/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.5111 - loss: 0.6922

155/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.5121 - loss: 0.6921

157/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.5128 - loss: 0.6919

159/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.5121 - loss: 0.6918

161/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.5120 - loss: 0.6918

163/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.5124 - loss: 0.6917

165/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.5132 - loss: 0.6917

167/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.5137 - loss: 0.6915

169/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.5141 - loss: 0.6915

171/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.5139 - loss: 0.6915

173/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.5128 - loss: 0.6916

175/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.5138 - loss: 0.6914

177/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.5140 - loss: 0.6912

179/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.5155 - loss: 0.6909

181/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.5167 - loss: 0.6904

183/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.5178 - loss: 0.6900

185/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.5176 - loss: 0.6916

187/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.5181 - loss: 0.6913

189/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.5203 - loss: 0.6911

191/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.5213 - loss: 0.6935

193/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.5217 - loss: 0.6937

195/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.5217 - loss: 0.6938

197/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.5222 - loss: 0.6936

199/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.5224 - loss: 0.6935

201/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.5217 - loss: 0.6934

203/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.5222 - loss: 0.6933

205/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.5223 - loss: 0.6932

207/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.5230 - loss: 0.6930

209/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.5237 - loss: 0.6929

211/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.5234 - loss: 0.6930

213/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.5234 - loss: 0.6929

215/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.5242 - loss: 0.6928

217/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5246 - loss: 0.6927

219/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5247 - loss: 0.6926

221/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5250 - loss: 0.6925

223/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5249 - loss: 0.6925

225/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5247 - loss: 0.6925

226/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5248 - loss: 0.6925

228/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5252 - loss: 0.6923

230/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5254 - loss: 0.6922

232/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5260 - loss: 0.6920

234/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5264 - loss: 0.6919

236/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5264 - loss: 0.6917

238/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5269 - loss: 0.6916

240/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.5272 - loss: 0.6914

242/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.5278 - loss: 0.6915

244/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.5281 - loss: 0.6916

246/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.5288 - loss: 0.6914

248/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.5298 - loss: 0.6912

250/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.5300 - loss: 0.6913

252/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.5301 - loss: 0.6913

254/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.5305 - loss: 0.6912

256/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.5312 - loss: 0.6910

258/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.5319 - loss: 0.6907

260/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.5318 - loss: 0.6907

262/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.5327 - loss: 0.6905

264/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.5334 - loss: 0.6902

266/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.5339 - loss: 0.6901

268/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.5346 - loss: 0.6898

270/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.5351 - loss: 0.6896

272/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.5355 - loss: 0.6896

274/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.5360 - loss: 0.6895

276/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.5366 - loss: 0.6895

278/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.5364 - loss: 0.6898

280/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.5372 - loss: 0.6894

282/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.5380 - loss: 0.6892

284/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.5385 - loss: 0.6890

286/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.5388 - loss: 0.6890

288/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.5392 - loss: 0.6890

290/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.5400 - loss: 0.6887

292/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.5400 - loss: 0.6889

294/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.5404 - loss: 0.6887

296/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.5407 - loss: 0.6886

298/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.5411 - loss: 0.6884

300/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.5411 - loss: 0.6883

302/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.5410 - loss: 0.6883

304/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.5413 - loss: 0.6881

306/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.5420 - loss: 0.6877

308/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.5427 - loss: 0.6873

310/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.5429 - loss: 0.6872

312/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.5436 - loss: 0.6868

313/313 ━━━━━━━━━━━━━━━━━━━━ 15s 44ms/step - accuracy: 0.5436 - loss: 0.6869 - val_accuracy: 0.6126 - val_loss: 0.6605


Epoch 2/5


  1/313 ━━━━━━━━━━━━━━━━━━━━ 17s 56ms/step - accuracy: 0.5781 - loss: 0.6352

  3/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.5990 - loss: 0.6576

  5/313 ━━━━━━━━━━━━━━━━━━━━ 12s 40ms/step - accuracy: 0.6094 - loss: 0.6597

  7/313 ━━━━━━━━━━━━━━━━━━━━ 12s 40ms/step - accuracy: 0.6161 - loss: 0.6571

  9/313 ━━━━━━━━━━━━━━━━━━━━ 12s 40ms/step - accuracy: 0.5938 - loss: 0.6735

 11/313 ━━━━━━━━━━━━━━━━━━━━ 12s 40ms/step - accuracy: 0.5980 - loss: 0.6692

 13/313 ━━━━━━━━━━━━━━━━━━━━ 12s 40ms/step - accuracy: 0.6070 - loss: 0.6647

 15/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6146 - loss: 0.6610

 17/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6213 - loss: 0.6592

 19/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6176 - loss: 0.6608

 21/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6220 - loss: 0.6601

 23/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6250 - loss: 0.6590

 25/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6319 - loss: 0.6562

 27/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6348 - loss: 0.6531

 29/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6352 - loss: 0.6532

 31/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6401 - loss: 0.6499

 33/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6406 - loss: 0.6508

 35/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6433 - loss: 0.6501

 37/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6385 - loss: 0.6564

 39/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6350 - loss: 0.6593

 41/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6284 - loss: 0.6619

 43/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6195 - loss: 0.6649

 45/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6174 - loss: 0.6659

 47/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6110 - loss: 0.6680

 49/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6062 - loss: 0.6697

 51/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6002 - loss: 0.6717

 53/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.5964 - loss: 0.6729

 55/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.5926 - loss: 0.6738

 57/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.5894 - loss: 0.6747

 59/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.5847 - loss: 0.6755

 61/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.5807 - loss: 0.6763

 63/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5786 - loss: 0.6770 

 65/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5764 - loss: 0.6774

 66/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5750 - loss: 0.6776

 68/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5719 - loss: 0.6780

 70/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5710 - loss: 0.6784

 72/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5688 - loss: 0.6788

 74/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.5676 - loss: 0.6791

 76/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5662 - loss: 0.6794

 78/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5669 - loss: 0.6794

 80/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5639 - loss: 0.6798

 82/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5604 - loss: 0.6804

 84/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5595 - loss: 0.6807

 86/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5612 - loss: 0.6806

 88/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5609 - loss: 0.6807

 90/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.5608 - loss: 0.6809

 92/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.5596 - loss: 0.6812

 94/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.5578 - loss: 0.6815

 96/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.5581 - loss: 0.6817

 98/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.5585 - loss: 0.6817

100/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.5570 - loss: 0.6818

102/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.5564 - loss: 0.6818

104/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.5560 - loss: 0.6820

106/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.5553 - loss: 0.6820

108/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.5545 - loss: 0.6823

110/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.5540 - loss: 0.6823

112/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.5529 - loss: 0.6825

114/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.5522 - loss: 0.6827

116/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.5521 - loss: 0.6828

118/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.5520 - loss: 0.6829

120/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.5512 - loss: 0.6830

122/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.5507 - loss: 0.6832

124/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.5504 - loss: 0.6833

126/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.5511 - loss: 0.6834

128/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.5515 - loss: 0.6835

130/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.5510 - loss: 0.6836

132/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.5496 - loss: 0.6838

134/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.5479 - loss: 0.6839

136/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.5483 - loss: 0.6839

138/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.5477 - loss: 0.6841

140/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.5480 - loss: 0.6840

142/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.5480 - loss: 0.6840

144/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.5469 - loss: 0.6841

146/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.5477 - loss: 0.6840

148/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.5480 - loss: 0.6841

150/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.5480 - loss: 0.6841

152/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.5467 - loss: 0.6843

154/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.5463 - loss: 0.6842

156/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.5471 - loss: 0.6843

158/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.5472 - loss: 0.6841

160/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.5470 - loss: 0.6841

162/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.5456 - loss: 0.6845

164/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5447 - loss: 0.6846

166/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5461 - loss: 0.6843

168/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5459 - loss: 0.6842

170/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5457 - loss: 0.6840

172/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5457 - loss: 0.6840

174/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5463 - loss: 0.6837

176/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5465 - loss: 0.6836

178/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5457 - loss: 0.6835

180/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5464 - loss: 0.6833

182/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5469 - loss: 0.6831

184/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5468 - loss: 0.6830

186/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5462 - loss: 0.6830

188/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.5450 - loss: 0.6830

190/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.5447 - loss: 0.6829

192/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.5447 - loss: 0.6827

194/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.5449 - loss: 0.6826

196/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.5451 - loss: 0.6823

198/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.5454 - loss: 0.6821

200/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.5450 - loss: 0.6820

202/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.5459 - loss: 0.6818

204/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.5465 - loss: 0.6817

206/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.5470 - loss: 0.6814

208/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.5460 - loss: 0.6815

210/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.5456 - loss: 0.6815

212/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.5461 - loss: 0.6815

214/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5465 - loss: 0.6813

216/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5462 - loss: 0.6815

218/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5465 - loss: 0.6814

220/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5467 - loss: 0.6813

222/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5467 - loss: 0.6812

224/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5467 - loss: 0.6810

226/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5470 - loss: 0.6809

228/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5473 - loss: 0.6807

230/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5482 - loss: 0.6804

232/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5485 - loss: 0.6802

234/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5488 - loss: 0.6801

236/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5493 - loss: 0.6799

238/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.5500 - loss: 0.6795

240/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.5503 - loss: 0.6793

242/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.5507 - loss: 0.6790

244/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.5507 - loss: 0.6791

246/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.5507 - loss: 0.6789

248/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.5515 - loss: 0.6784

250/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.5519 - loss: 0.6784

252/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.5525 - loss: 0.6783

254/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.5531 - loss: 0.6781

256/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.5526 - loss: 0.6780

258/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.5527 - loss: 0.6779

260/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.5534 - loss: 0.6775

262/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.5535 - loss: 0.6774

264/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5529 - loss: 0.6774

266/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5534 - loss: 0.6770

268/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5536 - loss: 0.6769

270/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5541 - loss: 0.6765

272/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5547 - loss: 0.6761

274/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5552 - loss: 0.6759

276/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5557 - loss: 0.6756

278/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5560 - loss: 0.6754

280/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5560 - loss: 0.6754

282/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5561 - loss: 0.6751

284/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5569 - loss: 0.6747

286/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5567 - loss: 0.6746

288/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.5565 - loss: 0.6744

290/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.5575 - loss: 0.6741

292/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.5581 - loss: 0.6740

294/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.5582 - loss: 0.6739

296/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.5585 - loss: 0.6736

298/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.5593 - loss: 0.6731

300/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.5594 - loss: 0.6729

302/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.5599 - loss: 0.6728

304/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.5611 - loss: 0.6723

306/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.5612 - loss: 0.6720

308/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.5613 - loss: 0.6721

310/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.5618 - loss: 0.6716

312/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.5619 - loss: 0.6715

313/313 ━━━━━━━━━━━━━━━━━━━━ 14s 43ms/step - accuracy: 0.5620 - loss: 0.6713 - val_accuracy: 0.6128 - val_loss: 0.6359


Epoch 3/5


  1/313 ━━━━━━━━━━━━━━━━━━━━ 17s 56ms/step - accuracy: 0.6875 - loss: 0.5634

  3/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.6250 - loss: 0.6062

  5/313 ━━━━━━━━━━━━━━━━━━━━ 12s 40ms/step - accuracy: 0.6438 - loss: 0.6135

  7/313 ━━━━━━━━━━━━━━━━━━━━ 12s 40ms/step - accuracy: 0.6317 - loss: 0.6136

  9/313 ━━━━━━━━━━━━━━━━━━━━ 12s 40ms/step - accuracy: 0.6372 - loss: 0.6178

 11/313 ━━━━━━━━━━━━━━━━━━━━ 12s 40ms/step - accuracy: 0.6278 - loss: 0.6184

 13/313 ━━━━━━━━━━━━━━━━━━━━ 12s 40ms/step - accuracy: 0.6274 - loss: 0.6187

 15/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6135 - loss: 0.6178

 17/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6131 - loss: 0.6167

 19/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6184 - loss: 0.6157

 21/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6205 - loss: 0.6127

 23/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6236 - loss: 0.6092

 25/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6275 - loss: 0.6100

 27/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6273 - loss: 0.6110

 29/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6315 - loss: 0.6086

 31/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6341 - loss: 0.6072

 33/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6321 - loss: 0.6050

 35/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6313 - loss: 0.6033

 37/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6313 - loss: 0.6016

 39/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6330 - loss: 0.6002

 41/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6315 - loss: 0.6012

 43/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6359 - loss: 0.5982

 45/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6347 - loss: 0.5978

 47/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6386 - loss: 0.5947

 49/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6378 - loss: 0.5942

 51/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6388 - loss: 0.5922

 53/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6415 - loss: 0.5894

 55/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6446 - loss: 0.5896

 57/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6469 - loss: 0.5903

 59/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6517 - loss: 0.5901

 61/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6545 - loss: 0.5881

 63/313 ━━━━━━━━━━━━━━━━━━━━ 10s 40ms/step - accuracy: 0.6555 - loss: 0.5919

 65/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6584 - loss: 0.5926 

 67/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6609 - loss: 0.5933

 69/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6624 - loss: 0.5942

 71/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6646 - loss: 0.5969

 73/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6657 - loss: 0.6000

 75/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6648 - loss: 0.6023

 77/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6658 - loss: 0.6031

 79/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6653 - loss: 0.6043

 81/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6647 - loss: 0.6054

 83/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6625 - loss: 0.6068

 85/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6616 - loss: 0.6076

 87/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6589 - loss: 0.6091

 89/313 ━━━━━━━━━━━━━━━━━━━━ 9s 40ms/step - accuracy: 0.6578 - loss: 0.6102

 91/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.6571 - loss: 0.6109

 93/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.6581 - loss: 0.6111

 95/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.6587 - loss: 0.6116

 97/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.6583 - loss: 0.6122

 99/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.6600 - loss: 0.6121

101/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.6600 - loss: 0.6125

103/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.6608 - loss: 0.6121

105/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.6598 - loss: 0.6123

107/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.6599 - loss: 0.6117

109/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.6610 - loss: 0.6118

111/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.6616 - loss: 0.6109

113/313 ━━━━━━━━━━━━━━━━━━━━ 8s 40ms/step - accuracy: 0.6619 - loss: 0.6110

115/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6643 - loss: 0.6098

117/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6669 - loss: 0.6084

119/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6685 - loss: 0.6077

121/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6692 - loss: 0.6068

123/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6704 - loss: 0.6060

125/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6719 - loss: 0.6053

127/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6714 - loss: 0.6057

129/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6733 - loss: 0.6044

131/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6749 - loss: 0.6032

133/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6762 - loss: 0.6023

135/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6771 - loss: 0.6023

137/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6770 - loss: 0.6024

139/313 ━━━━━━━━━━━━━━━━━━━━ 7s 40ms/step - accuracy: 0.6783 - loss: 0.6010

141/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.6790 - loss: 0.6006

143/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.6809 - loss: 0.5987

145/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.6821 - loss: 0.5973

147/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.6832 - loss: 0.5963

149/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.6844 - loss: 0.5959

151/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.6851 - loss: 0.5956

153/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.6854 - loss: 0.5967

155/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.6854 - loss: 0.5979

157/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.6853 - loss: 0.5998

159/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.6848 - loss: 0.6016

161/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.6842 - loss: 0.6028

163/313 ━━━━━━━━━━━━━━━━━━━━ 6s 40ms/step - accuracy: 0.6829 - loss: 0.6044

165/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6822 - loss: 0.6053

167/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6821 - loss: 0.6055

169/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6806 - loss: 0.6063

171/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6801 - loss: 0.6069

173/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6796 - loss: 0.6071

175/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6781 - loss: 0.6079

177/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6762 - loss: 0.6090

179/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6741 - loss: 0.6100

181/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6734 - loss: 0.6106

183/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6730 - loss: 0.6113

185/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6707 - loss: 0.6123

187/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6690 - loss: 0.6133

189/313 ━━━━━━━━━━━━━━━━━━━━ 5s 40ms/step - accuracy: 0.6677 - loss: 0.6140

191/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.6662 - loss: 0.6148

193/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.6654 - loss: 0.6156

195/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.6647 - loss: 0.6161

197/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.6639 - loss: 0.6168

199/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.6630 - loss: 0.6175

201/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.6621 - loss: 0.6181

203/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.6613 - loss: 0.6186

205/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.6595 - loss: 0.6194

207/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.6582 - loss: 0.6203

209/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.6573 - loss: 0.6210

211/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.6564 - loss: 0.6215

213/313 ━━━━━━━━━━━━━━━━━━━━ 4s 40ms/step - accuracy: 0.6549 - loss: 0.6222

215/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.6547 - loss: 0.6226

217/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.6543 - loss: 0.6230

219/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.6534 - loss: 0.6234

221/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.6525 - loss: 0.6239

223/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.6521 - loss: 0.6242

225/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.6510 - loss: 0.6249

227/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.6505 - loss: 0.6253

229/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.6496 - loss: 0.6257

231/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.6491 - loss: 0.6261

233/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.6480 - loss: 0.6269

235/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.6481 - loss: 0.6270

237/313 ━━━━━━━━━━━━━━━━━━━━ 3s 40ms/step - accuracy: 0.6477 - loss: 0.6273

239/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6470 - loss: 0.6276

241/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6468 - loss: 0.6278

243/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6456 - loss: 0.6285

245/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6453 - loss: 0.6287

247/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6446 - loss: 0.6293

249/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6443 - loss: 0.6294

251/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6436 - loss: 0.6297

253/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6428 - loss: 0.6303

255/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6422 - loss: 0.6308

257/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6420 - loss: 0.6312

259/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6415 - loss: 0.6313

261/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6409 - loss: 0.6316

263/313 ━━━━━━━━━━━━━━━━━━━━ 2s 40ms/step - accuracy: 0.6405 - loss: 0.6317

265/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.6403 - loss: 0.6319

267/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.6400 - loss: 0.6323

269/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.6392 - loss: 0.6326

271/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.6388 - loss: 0.6326

273/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.6391 - loss: 0.6327

275/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.6394 - loss: 0.6325

277/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.6393 - loss: 0.6328

279/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.6384 - loss: 0.6333

281/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.6383 - loss: 0.6334

283/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.6383 - loss: 0.6335

285/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.6375 - loss: 0.6336

287/313 ━━━━━━━━━━━━━━━━━━━━ 1s 40ms/step - accuracy: 0.6373 - loss: 0.6337

289/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6369 - loss: 0.6339

291/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6369 - loss: 0.6340

293/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6367 - loss: 0.6340

295/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6369 - loss: 0.6340

297/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6368 - loss: 0.6341

299/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6367 - loss: 0.6343

301/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6365 - loss: 0.6343

303/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6365 - loss: 0.6343

305/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6362 - loss: 0.6343

307/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6360 - loss: 0.6344

309/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6356 - loss: 0.6345

311/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6359 - loss: 0.6346

313/313 ━━━━━━━━━━━━━━━━━━━━ 0s 40ms/step - accuracy: 0.6356 - loss: 0.6346

313/313 ━━━━━━━━━━━━━━━━━━━━ 14s 43ms/step - accuracy: 0.6356 - loss: 0.6346 - val_accuracy: 0.6556 - val_loss: 0.6487


Epoch 4/5


  1/313 ━━━━━━━━━━━━━━━━━━━━ 16s 52ms/step - accuracy: 0.6094 - loss: 0.6059

  3/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.6354 - loss: 0.6070

  5/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.6313 - loss: 0.6185

  7/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.6406 - loss: 0.6151

  9/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.6597 - loss: 0.6069

 11/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.6591 - loss: 0.6098

 13/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.6635 - loss: 0.6112

 15/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.6635 - loss: 0.6152

 17/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.6654 - loss: 0.6201

 19/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.6620 - loss: 0.6192

 21/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.6644 - loss: 0.6154

 23/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.6671 - loss: 0.6135

 25/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.6706 - loss: 0.6124

 27/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.6690 - loss: 0.6130

 29/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.6751 - loss: 0.6097

 31/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.6754 - loss: 0.6078

 33/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.6776 - loss: 0.6066

 35/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.6768 - loss: 0.6093

 37/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6774 - loss: 0.6091

 39/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.6807 - loss: 0.6080

 41/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.6845 - loss: 0.6058

 43/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.6871 - loss: 0.6036

 45/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.6885 - loss: 0.6017

 46/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.6885 - loss: 0.6025

 48/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.6904 - loss: 0.6009

 50/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.6925 - loss: 0.5980

 52/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.6947 - loss: 0.5957

 54/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.6965 - loss: 0.5946

 56/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.6959 - loss: 0.5959

 58/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.6983 - loss: 0.5931

 60/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.6984 - loss: 0.5939

 62/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.6993 - loss: 0.5924

 64/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.7002 - loss: 0.5906

 66/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.7043 - loss: 0.5872

 68/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.7054 - loss: 0.5854

 70/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.7060 - loss: 0.5851 

 72/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.7059 - loss: 0.5843

 74/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.7080 - loss: 0.5828

 76/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.7072 - loss: 0.5843

 78/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.7061 - loss: 0.5844

 80/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.7076 - loss: 0.5830

 82/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.7075 - loss: 0.5815

 84/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.7094 - loss: 0.5791

 86/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.7095 - loss: 0.5795

 88/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.7100 - loss: 0.5778

 90/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.7118 - loss: 0.5769

 92/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.7138 - loss: 0.5751

 94/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.7148 - loss: 0.5758

 96/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.7153 - loss: 0.5749

 98/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.7165 - loss: 0.5730

100/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.7177 - loss: 0.5711

102/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.7195 - loss: 0.5693

104/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.7207 - loss: 0.5679

106/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.7216 - loss: 0.5673

108/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.7227 - loss: 0.5667

110/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.7243 - loss: 0.5647

112/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.7260 - loss: 0.5627

114/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.7267 - loss: 0.5622

116/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.7284 - loss: 0.5599

118/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.7293 - loss: 0.5591

120/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.7293 - loss: 0.5591

122/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.7295 - loss: 0.5589

124/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.7300 - loss: 0.5580

126/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.7310 - loss: 0.5562

128/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.7311 - loss: 0.5558

130/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.7320 - loss: 0.5550

132/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.7325 - loss: 0.5538

134/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.7331 - loss: 0.5537

136/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.7344 - loss: 0.5516

138/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.7355 - loss: 0.5500

140/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.7364 - loss: 0.5483

142/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7361 - loss: 0.5481

144/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7374 - loss: 0.5469

146/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7390 - loss: 0.5452

148/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7406 - loss: 0.5431

150/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7411 - loss: 0.5431

152/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7422 - loss: 0.5417

154/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7428 - loss: 0.5405

156/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7432 - loss: 0.5403

158/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7441 - loss: 0.5392

159/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7444 - loss: 0.5386

160/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7445 - loss: 0.5386

161/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7451 - loss: 0.5378

162/313 ━━━━━━━━━━━━━━━━━━━━ 6s 42ms/step - accuracy: 0.7458 - loss: 0.5371

164/313 ━━━━━━━━━━━━━━━━━━━━ 6s 42ms/step - accuracy: 0.7463 - loss: 0.5363

166/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7468 - loss: 0.5359

168/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.7474 - loss: 0.5356

170/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.7473 - loss: 0.5361

172/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.7470 - loss: 0.5366

174/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.7478 - loss: 0.5355

176/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.7482 - loss: 0.5347

178/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.7489 - loss: 0.5339

180/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.7495 - loss: 0.5335

182/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.7500 - loss: 0.5331

184/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.7504 - loss: 0.5325

186/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.7510 - loss: 0.5321

188/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.7512 - loss: 0.5321

190/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.7517 - loss: 0.5317

192/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.7522 - loss: 0.5313

194/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.7522 - loss: 0.5309

196/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.7521 - loss: 0.5313

198/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.7524 - loss: 0.5307

200/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.7527 - loss: 0.5303

202/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.7532 - loss: 0.5296

204/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.7537 - loss: 0.5289

206/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.7546 - loss: 0.5279

208/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.7550 - loss: 0.5277

210/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.7557 - loss: 0.5268

212/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.7555 - loss: 0.5265

214/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.7561 - loss: 0.5255

216/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.7560 - loss: 0.5260

218/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.7565 - loss: 0.5251

220/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.7565 - loss: 0.5252

222/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.7569 - loss: 0.5253

224/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.7572 - loss: 0.5252

226/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.7581 - loss: 0.5237

228/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.7585 - loss: 0.5230

230/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.7578 - loss: 0.5243

232/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.7573 - loss: 0.5248

234/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.7579 - loss: 0.5242

236/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.7583 - loss: 0.5235

238/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.7590 - loss: 0.5229

240/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.7593 - loss: 0.5225

242/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.7596 - loss: 0.5219

244/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.7588 - loss: 0.5228

246/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.7588 - loss: 0.5230

248/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.7593 - loss: 0.5223

250/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.7595 - loss: 0.5223

252/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.7605 - loss: 0.5213

254/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.7606 - loss: 0.5212

256/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.7610 - loss: 0.5209

258/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.7611 - loss: 0.5208

260/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.7614 - loss: 0.5204

262/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.7614 - loss: 0.5203

264/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.7616 - loss: 0.5201

266/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.7619 - loss: 0.5195

268/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.7621 - loss: 0.5189

270/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.7630 - loss: 0.5179

272/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.7635 - loss: 0.5175

274/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.7633 - loss: 0.5176

276/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.7635 - loss: 0.5176

278/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.7635 - loss: 0.5174

280/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.7640 - loss: 0.5170

282/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.7642 - loss: 0.5167

284/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.7641 - loss: 0.5171

286/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.7645 - loss: 0.5167

288/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.7651 - loss: 0.5162

290/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.7652 - loss: 0.5159

292/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.7656 - loss: 0.5153

294/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.7656 - loss: 0.5152

296/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.7660 - loss: 0.5145

298/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.7666 - loss: 0.5140

300/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.7669 - loss: 0.5136

302/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.7673 - loss: 0.5129

304/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.7672 - loss: 0.5131

306/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.7672 - loss: 0.5131

308/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.7674 - loss: 0.5127

310/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.7678 - loss: 0.5123

312/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.7681 - loss: 0.5118

313/313 ━━━━━━━━━━━━━━━━━━━━ 14s 45ms/step - accuracy: 0.7682 - loss: 0.5116 - val_accuracy: 0.8064 - val_loss: 0.4518


Epoch 5/5


  1/313 ━━━━━━━━━━━━━━━━━━━━ 18s 58ms/step - accuracy: 0.8438 - loss: 0.3911

  3/313 ━━━━━━━━━━━━━━━━━━━━ 12s 42ms/step - accuracy: 0.8281 - loss: 0.4345

  5/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.8313 - loss: 0.4365

  7/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.8393 - loss: 0.4237

  9/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.8455 - loss: 0.4143

 11/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.8452 - loss: 0.4017

 13/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.8353 - loss: 0.4118

 15/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.8323 - loss: 0.4152

 17/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.8382 - loss: 0.4044

 19/313 ━━━━━━━━━━━━━━━━━━━━ 12s 41ms/step - accuracy: 0.8363 - loss: 0.4065

 21/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.8318 - loss: 0.4079

 23/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.8295 - loss: 0.4140

 25/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.8313 - loss: 0.4129

 27/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.8270 - loss: 0.4206

 29/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.8260 - loss: 0.4183

 31/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.8291 - loss: 0.4150

 33/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.8314 - loss: 0.4117

 35/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.8330 - loss: 0.4098

 37/313 ━━━━━━━━━━━━━━━━━━━━ 11s 40ms/step - accuracy: 0.8311 - loss: 0.4106

 39/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.8317 - loss: 0.4115

 41/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.8300 - loss: 0.4153

 43/313 ━━━━━━━━━━━━━━━━━━━━ 11s 41ms/step - accuracy: 0.8310 - loss: 0.4145

 45/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.8299 - loss: 0.4137

 47/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.8291 - loss: 0.4166

 49/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.8281 - loss: 0.4160

 51/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.8278 - loss: 0.4162

 53/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.8275 - loss: 0.4161

 55/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.8290 - loss: 0.4148

 57/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.8281 - loss: 0.4160

 59/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.8294 - loss: 0.4138

 61/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.8274 - loss: 0.4166

 63/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.8269 - loss: 0.4174

 65/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.8267 - loss: 0.4169

 67/313 ━━━━━━━━━━━━━━━━━━━━ 10s 41ms/step - accuracy: 0.8288 - loss: 0.4143

 69/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8295 - loss: 0.4130 

 71/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8292 - loss: 0.4141

 73/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8294 - loss: 0.4137

 75/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8298 - loss: 0.4131

 77/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8302 - loss: 0.4119

 79/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8309 - loss: 0.4111

 81/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8293 - loss: 0.4124

 83/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8294 - loss: 0.4120

 85/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8283 - loss: 0.4129

 87/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8285 - loss: 0.4116

 89/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8290 - loss: 0.4113

 91/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8281 - loss: 0.4123

 93/313 ━━━━━━━━━━━━━━━━━━━━ 9s 41ms/step - accuracy: 0.8281 - loss: 0.4132

 95/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.8285 - loss: 0.4126

 97/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.8289 - loss: 0.4111

 99/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.8280 - loss: 0.4119

101/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.8267 - loss: 0.4139

103/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.8271 - loss: 0.4130

105/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.8272 - loss: 0.4129

107/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.8271 - loss: 0.4143

109/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.8271 - loss: 0.4143

111/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.8270 - loss: 0.4146

113/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.8280 - loss: 0.4131

115/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.8285 - loss: 0.4124

116/313 ━━━━━━━━━━━━━━━━━━━━ 8s 41ms/step - accuracy: 0.8292 - loss: 0.4116

118/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8288 - loss: 0.4118

120/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8289 - loss: 0.4120

122/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8285 - loss: 0.4123

124/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8281 - loss: 0.4135

126/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8281 - loss: 0.4130

128/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8282 - loss: 0.4125

130/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8299 - loss: 0.4105

132/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8286 - loss: 0.4119

134/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8288 - loss: 0.4116

136/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8289 - loss: 0.4122

138/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8287 - loss: 0.4122

140/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8291 - loss: 0.4110

142/313 ━━━━━━━━━━━━━━━━━━━━ 7s 41ms/step - accuracy: 0.8293 - loss: 0.4108

144/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.8294 - loss: 0.4101

146/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.8295 - loss: 0.4099

148/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.8302 - loss: 0.4084

150/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.8308 - loss: 0.4070

152/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.8311 - loss: 0.4063

154/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.8312 - loss: 0.4060

156/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.8316 - loss: 0.4056

158/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.8317 - loss: 0.4053

160/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.8319 - loss: 0.4047

162/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.8324 - loss: 0.4035

164/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.8325 - loss: 0.4033

166/313 ━━━━━━━━━━━━━━━━━━━━ 6s 41ms/step - accuracy: 0.8327 - loss: 0.4026

168/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.8325 - loss: 0.4026

170/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.8335 - loss: 0.4013

172/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.8339 - loss: 0.4003

174/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.8341 - loss: 0.4003

176/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.8346 - loss: 0.3996

178/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.8354 - loss: 0.3981

180/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.8360 - loss: 0.3970

182/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.8359 - loss: 0.3968

184/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.8361 - loss: 0.3966

186/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.8364 - loss: 0.3966

188/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.8362 - loss: 0.3970

190/313 ━━━━━━━━━━━━━━━━━━━━ 5s 41ms/step - accuracy: 0.8365 - loss: 0.3964

192/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.8366 - loss: 0.3961

194/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.8368 - loss: 0.3961

196/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.8371 - loss: 0.3952

198/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.8375 - loss: 0.3950

200/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.8379 - loss: 0.3947

202/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.8385 - loss: 0.3935

204/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.8387 - loss: 0.3931

206/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.8394 - loss: 0.3917

208/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.8391 - loss: 0.3919

210/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.8390 - loss: 0.3924

212/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.8393 - loss: 0.3914

214/313 ━━━━━━━━━━━━━━━━━━━━ 4s 41ms/step - accuracy: 0.8392 - loss: 0.3920

216/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.8395 - loss: 0.3915

218/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.8400 - loss: 0.3904

220/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.8396 - loss: 0.3911

222/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.8396 - loss: 0.3906

224/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.8398 - loss: 0.3902

226/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.8392 - loss: 0.3913

228/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.8390 - loss: 0.3916

230/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.8395 - loss: 0.3910

232/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.8398 - loss: 0.3911

234/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.8393 - loss: 0.3919

236/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.8394 - loss: 0.3921

238/313 ━━━━━━━━━━━━━━━━━━━━ 3s 41ms/step - accuracy: 0.8392 - loss: 0.3923

240/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8393 - loss: 0.3922

242/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8396 - loss: 0.3921

244/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8394 - loss: 0.3923

246/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8389 - loss: 0.3932

248/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8381 - loss: 0.3942

250/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8375 - loss: 0.3952

252/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8374 - loss: 0.3955

254/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8377 - loss: 0.3949

256/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8378 - loss: 0.3948

258/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8375 - loss: 0.3954

260/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8371 - loss: 0.3960

262/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8374 - loss: 0.3957

264/313 ━━━━━━━━━━━━━━━━━━━━ 2s 41ms/step - accuracy: 0.8377 - loss: 0.3954

266/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.8377 - loss: 0.3956

268/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.8379 - loss: 0.3954

270/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.8378 - loss: 0.3953

272/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.8379 - loss: 0.3953

274/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.8380 - loss: 0.3951

276/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.8381 - loss: 0.3948

278/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.8380 - loss: 0.3953

280/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.8378 - loss: 0.3959

282/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.8380 - loss: 0.3958

284/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.8381 - loss: 0.3957

286/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.8380 - loss: 0.3962

288/313 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - accuracy: 0.8382 - loss: 0.3960

290/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.8381 - loss: 0.3959

292/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.8379 - loss: 0.3961

294/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.8367 - loss: 0.3974

296/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.8363 - loss: 0.3976

298/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.8361 - loss: 0.3979

300/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.8353 - loss: 0.3988

302/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.8350 - loss: 0.3992

304/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.8338 - loss: 0.4005

306/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.8337 - loss: 0.4006

308/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.8335 - loss: 0.4009

310/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.8337 - loss: 0.4005

312/313 ━━━━━━━━━━━━━━━━━━━━ 0s 41ms/step - accuracy: 0.8334 - loss: 0.4008

313/313 ━━━━━━━━━━━━━━━━━━━━ 14s 44ms/step - accuracy: 0.8333 - loss: 0.4011 - val_accuracy: 0.7818 - val_loss: 0.4575


## Section 5: Evaluate with Precision, Recall, F1-score

In [6]:
y_prob = model.predict(X_test, verbose=0).ravel()
y_pred = (y_prob >= 0.5).astype(int)

cm = confusion_matrix(y_test, y_pred)
print("Confusion matrix:")
print(cm)

print("\nClassification report:")
print(
    classification_report(
        y_test, y_pred, target_names=["Negative", "Positive"]
    )
)

report_dict = classification_report(
    y_test,
    y_pred,
    target_names=["Negative", "Positive"],
    output_dict=True,
)
report_df = pd.DataFrame(report_dict).transpose()
report_csv = results_dir / "classification_report.csv"
report_df.to_csv(report_csv)
print("Saved:", report_csv)

overall_test_acc = float((y_pred == y_test).mean())
print(f"Overall Test Accuracy: {overall_test_acc * 100:.2f}%")

Confusion matrix:
[[ 9077  3423]
 [ 2182 10318]]

Classification report:
              precision    recall  f1-score   support

    Negative       0.81      0.73      0.76     12500
    Positive       0.75      0.83      0.79     12500

    accuracy                           0.78     25000
   macro avg       0.78      0.78      0.78     25000
weighted avg       0.78      0.78      0.78     25000

Saved: /Users/yuvrajgole/Documents/DL Experiments/DL Experiment 5/nlp-imdb-lstm/results/classification_report.csv
Overall Test Accuracy: 77.58%


## Section 6: Visualize Results

In [7]:
fig, axes = plt.subplots(1, 2, figsize=(12, 5))

axes[0].plot(history.history["accuracy"], label="Train")
axes[0].plot(history.history["val_accuracy"], label="Validation")
axes[0].set_title("Accuracy")
axes[0].set_xlabel("Epoch")
axes[0].set_ylabel("Accuracy")
axes[0].legend()
axes[0].grid(alpha=0.3)

axes[1].plot(history.history["loss"], label="Train")
axes[1].plot(history.history["val_loss"], label="Validation")
axes[1].set_title("Loss")
axes[1].set_xlabel("Epoch")
axes[1].set_ylabel("Loss")
axes[1].legend()
axes[1].grid(alpha=0.3)

fig.suptitle("Training vs Validation Curves")
fig.tight_layout()
curves_path = results_dir / "training_curves.png"
fig.savefig(curves_path, dpi=120, bbox_inches="tight")
plt.close(fig)
print("Saved:", curves_path)

fig, ax = plt.subplots(figsize=(6, 5))
sns.heatmap(
    cm,
    annot=True,
    fmt="d",
    cmap="Blues",
    xticklabels=["Negative", "Positive"],
    yticklabels=["Negative", "Positive"],
    ax=ax,
)
ax.set_xlabel("Predicted")
ax.set_ylabel("Actual")
ax.set_title("Confusion Matrix")
fig.tight_layout()
cm_path = results_dir / "confusion_matrix.png"
fig.savefig(cm_path, dpi=120, bbox_inches="tight")
plt.close(fig)
print("Saved:", cm_path)

Saved: /Users/yuvrajgole/Documents/DL Experiments/DL Experiment 5/nlp-imdb-lstm/results/training_curves.png


Saved: /Users/yuvrajgole/Documents/DL Experiments/DL Experiment 5/nlp-imdb-lstm/results/confusion_matrix.png


## Section 7: Save Model

In [8]:
model_path = models_dir / "imdb_lstm.keras"
model.save(model_path)

print(f"Saved model to {model_path} (Keras .keras format)")
print("Text Preprocessed and Padded")
print("Embedding + LSTM Model Trained")
print(f"Overall Test Accuracy: {overall_test_acc * 100:.2f}%")
print("Evaluation Metrics Generated")

Saved model to /Users/yuvrajgole/Documents/DL Experiments/DL Experiment 5/nlp-imdb-lstm/models/imdb_lstm.keras (Keras .keras format)
Text Preprocessed and Padded
Embedding + LSTM Model Trained
Overall Test Accuracy: 77.58%
Evaluation Metrics Generated


## Section 8: Results Summary

In [9]:
summary_table = pd.DataFrame(
    {
        "Class": ["Negative", "Positive"],
        "Precision": [
            report_dict["Negative"]["precision"],
            report_dict["Positive"]["precision"],
        ],
        "Recall": [
            report_dict["Negative"]["recall"],
            report_dict["Positive"]["recall"],
        ],
        "F1-Score": [
            report_dict["Negative"]["f1-score"],
            report_dict["Positive"]["f1-score"],
        ],
    }
)
print(summary_table.to_string(index=False))
summary_table

   Class  Precision  Recall  F1-Score
Negative   0.806199 0.72616  0.764089
Positive   0.750891 0.82544  0.786403


,Class,Precision,Recall,F1-Score
0,Negative,0.806199,0.72616,0.764089
1,Positive,0.750891,0.82544,0.786403


### Completion Checklist

- [x] Text Preprocessed and Padded
- [x] Embedding + LSTM Model Trained
- [x] Overall Test Accuracy: 77.58%
- [x] Evaluation Metrics Generated